sdks/java/core/src/main/java/org/apache/beam/sdk/io/Source.java

```diff
@@ -63,6 +63,7 @@
 
   /**
    * Returns the default {@code Coder} to use for the data read from this source.
+   * @throws org.apache.beam.sdk.coders.CannotProvideCoderException if there is no default Coder.
```

Comment:
It is a checked exception, so I think you'd also need to change the interface here for this to be true, no?

Generally it seems correct that this should be able to throw this exception.